### PR TITLE
Feature/#347 implement follow followers in profile

### LINKED ISSUE #347: Implement Follow / Followers in profile
As a user I would like to see the number of followers / people I follow + the ability to follow another user

NOTE:
- forget the 'search for investor' feature, DONT do it, no time and doesn't bring that much value

**Acceptance Criteria**
- Show the number of followers / followees they have
- If the user is not viewing their own profile show the action button "Follow / Unfollow"
- When clicking on the followers / followees button, show a popup -> when clicking a user in the popup, navigate to their profile
- In the popup, sort investors alphabetically

### PR DESCRIPTION
**Pull request by EspressoCode**
# Related Issue
- closes #347 Implement Follow / Followers in profile

# Proposed changes
- added followers/followings count in profile
- added follow/unfollow button and the follow/unfollow functionality

# Reviewer(s)
 - @Alessandro100 @arthur-del

### CODE REVIEW

File: client/src/app/components/list-of-follows/list-of-follows.component.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    public dialogRef: MatDialogRef<ListOfFollowsComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: any
+  ) {
+    this.follows = data.followslist;
+    this.listTitle = data.followsTitle;
+  }
+
+  print(): void {
+    console.log(this.follows);
+  }
```
Comment (Alessandro100) [inline client/src/app/components/list-of-follows/list-of-follows.component.ts:24]: We can remove this funciton

File: client/src/app/pages/profile/profile.component.html [hunk 2]
```diff
@@ -1,3 +1,16 @@
...
\ No newline at end of file
+
+</app-transaction-history>
+<div class="followers-follwings-container" *ngIf="completeUserProfile">
+
+    <button mat-button class="follow-unfollow-label" (click)="openDialog('followers')"
+        [disabled]="getFollowers().length===0">{{getFollowers().length}}
+        followers</button>
+    <button mat-button class="follow-unfollow-label" (click)="openDialog('followings')"
+        [disabled]="getFollowings().length===0">{{getFollowings().length}}
```
Comment (Alessandro100) [inline client/src/app/pages/profile/profile.component.html:11]: we shouldn't disable the button if they don't have any followers, we can open up the popup and say "hey you aren't following anyone"

**Review by Alessandro100 — APPROVED** [review]
Nice code! Just remove the [disable] if there are 0 followers

**Comment by EspressoCode** [comment]
Please try the branch one more time and see the changes I made. I removed userService from the profile component and now we are using storeFacade for when follow/unfollow button is clicked.

**Comment by Alessandro100** [comment]
Nicely done

**Pull request merged**